nukebuild/BuildParameters.cs

```diff
@@ -0,0 +1,186 @@
+                
+                // TODO???
+                IsTagged = false;
+                tagName = null;
+            }
+            IsMainRepo =
+                StringComparer.OrdinalIgnoreCase.Equals(MainRepo,
+                    RepositoryName);
+            IsMasterBranch = StringComparer.OrdinalIgnoreCase.Equals(MasterBranch,
+                RepositoryBranch);
+
+            
+            IsReleasable = StringComparer.OrdinalIgnoreCase.Equals(ReleaseConfiguration, Configuration);
+            IsMyGetRelease = !IsTagged && IsReleasable;
+            IsNuGetRelease = IsMainRepo && IsReleasable && IsReleaseBranch;
+
+            // VERSION
+            Version = b.NukeArgForceNugetVersion ?? GetVersion();
+
+            if (IsRunningOnAppVeyor)
+            {
+                string tagVersion = null;
+                if (IsTagged)
+                {
+                    var tag = tagName;
+                    var nugetReleasePrefix = "nuget-release-";
+                    IsNuGetRelease = IsTagged && IsReleasable && tag.StartsWith(nugetReleasePrefix);
+                    if (IsNuGetRelease)
+                        tagVersion = tag.Substring(nugetReleasePrefix.Length);
+                }
+
+                if (tagVersion != null)
+                {
+                    Version = tagVersion;
+                }
+                else
+                {
+                    // Use AssemblyVersion with Build as version
+                    Version += "-build" + Environment.GetEnvironmentVariable("APPVEYOR_BUILD_NUMBER") + "-beta";
+                }
+            }
+            else if (IsRunningOnAzure)
+            {
+                if (!IsNuGetRelease)
+                {
+                    // Use AssemblyVersion with Build as version
+                    Version += "-build" + Environment.GetEnvironmentVariable("BUILD_BUILDID") + "-beta";
+                }
+
+                PublishTestResults = true;
+            }
+
+            // DIRECTORIES
+            ArtifactsDir = RootDirectory / "artifacts";
+            NugetRoot = ArtifactsDir / "nuget";
+            ZipRoot = ArtifactsDir / "zip";
+            BinRoot = ArtifactsDir / "bin";
+            TestResultsRoot = ArtifactsDir / "test-results";
+            BuildDirs = GlobDirectories(RootDirectory, "**bin").Concat(GlobDirectories(RootDirectory, "**obj")).ToList();
+            DirSuffix = Configuration;
+            FileZipSuffix = Version + ".zip";
+            ZipCoreArtifacts = ZipRoot / ("Avalonia-" + FileZipSuffix);
+            ZipNuGetArtifacts = ZipRoot / ("Avalonia-NuGet-" + FileZipSuffix);
+            ZipSourceControlCatalogDesktopDir =
+                RootDirectory / ("samples/ControlCatalog.Desktop/bin/" + DirSuffix + "/net461");
+            ZipTargetControlCatalogDesktopDir = ZipRoot / ("ControlCatalog.Desktop-" + FileZipSuffix);
+        }
+
+        private static string GetVersion()
```

Comment:
I think we can re-write this to fetch the property from the props file via Nuke's ability to parse project files.

Reply:
For some reason it couldn't  inject solution via `[Solution]`. I think it's better to keep the rock-solid approach of reading XML directly.